echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/email/EmailNotificationService.groovy

```diff
@@ -55,12 +55,29 @@ class EmailNotificationService implements NotificationService {
 
   @Override
   EchoResponse.Void handle(Notification notification) {
-    def to = notification.to?.collect { it.split(" ") }?.flatten() as String[]
-    def cc = notification.cc?.collect { it.split(" ") }?.flatten() as String[]
+    def expando = { String addresses ->
+      // could be comma and/or space separated
+      addresses = addresses.replaceAll(",", " ")
+      return addresses.split(" ").findAll { !it.isEmpty() }
+    }
+
+    def to = notification.to?.collect {
+      expando(it)
+    }?.flatten() as String[]
+
+    def cc = notification.cc?.collect {
+      expando(it)
+    }?.flatten() as String[]
+
     def subject = notificationTemplateEngine.build(notification, NotificationTemplateEngine.Type.SUBJECT)
     def body = notificationTemplateEngine.build(notification, NotificationTemplateEngine.Type.BODY)
 
-    send(to, cc, subject, body)
+    try {
+      send(to, cc, subject, body)
+    } catch (AddressException e) {
+      throw new InvalidRequestException(e.message)
```

Comment:
should result in a 400 vs 500 coming back if a bad address is submitted ... won't trip our internal alerts